### PR TITLE
feat(workflow): Add pull request template and workflows to enforce new PR title rules

### PR DESCRIPTION
**Pull request by sihyung-maxim**
This PR adds guidelines and examples on titling pull requests within the MSDK repo that Jacob, Jake, and I came up with.

Sort of a brain dump at the moment. Needs more revising. Any additions and suggestions are welcome.

For the bug-related PRs, we could create a workflow that will automatically create a GitHub issue if the PR is resolving a Jira ticket.

### CODE REVIEW

**Comment by Jake-Carter** [comment]
@sihyung-maxim @Jacob-Scheiffler I simplified the template a bit and moved it to the `.github` folder.  I don't think it's worth overdoing it here on any actions/workflows.  This is a good enough start to avoid empty descriptions/titles

https://github.com/Analog-Devices-MSDK/msdk/pull/553/commits/cf2dca5074ca0f09feb05cb35f9994c7251a0108

**Comment by sihyung-maxim** [comment]
> @sihyung-maxim @Jacob-Scheiffler I simplified the template a bit and moved it to the `.github` folder. I don't think it's worth overdoing it here on any actions/workflows. This is a good enough start to avoid empty descriptions/titles
> 
> [cf2dca5](https://github.com/Analog-Devices-MSDK/msdk/commit/cf2dca5074ca0f09feb05cb35f9994c7251a0108)

I haven't pushed in my latest changes yet due to other tasks popping up, but I found a way to standardize our PR titles using conventional commits paired with semantic versioning. I have a workflow that enforces the PR title changes and there are tools that will autogenerate the changelog if we follow this new convention.

I have an email drafted, which describes the changes, that I need to send, so we can decide whether we want to implement this new convention or not. I need to do some more research on the autogeneration of the changelog first.

File: .github/workflows/PR_Title_Check.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            refactor
+            docs
+            style
+            test
+            perf
+            ci
+            build
+            revert
+
+          scopes: |
```
Comment (Jake-Carter) [inline .github/workflows/PR_Title_Check.yml:47]: Add an "Other" scope

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+6.  **CMSIS** – Any CMSIS/Register file changes
+7.  **MiscDrivers** – Any MiscDrivers changes (e.g. TFT Display, PB, LED, External Flash Drivers)
+8.  **PeriphDrivers** – Any changes in msdk/Libraries/PeriphDrivers
+9.  **Third-Party** – All third-party libraries (e.g. FreeRTOS, LVGL, lwIP)
+10. **ignore** – Small and quick miscellaneous fixes (should not be used often)
+11. **workflow** – Any GitHub workflow related changes
+
+NOTE: The scope is case sensitive and must match one of the listed scopes.
+
+### Responsibilites for Reviewers/Mergers/PR Owners
```
Comment (Jake-Carter) [inline .github/pull_request_template.md:47]: I suggest moving this into CONTRIBUTING.md.  It makes the PR template too cluttered/long.  Keep in mind that users will need to delete these types of sections from the PR every time

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
+## Pull Request Template
+
+### Overview
```
Comment (Jake-Carter) [inline .github/pull_request_template.md:3]: Change this section to `### PR Guidelines`, add a template spot for the user's Overview content at the bottom.

File: .husky/commit-msg
Comment (Jake-Carter) [inline .husky/commit-msg:1]: What is this for?

**Review by Jake-Carter — CHANGES_REQUESTED** [review]

File: .husky/commit-msg
Comment (sihyung-maxim) [reply .husky/commit-msg:1]: Preconfigured git hooks in case you want to install the Commitlint tool locally onto your machine.

**Review by Jake-Carter — APPROVED** [review]

**Review by Jacob-Scheiffler — APPROVED** [review]

File: .github/labeler.yml [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+'MAX32690':
+- 'Examples/MAX32690/*'
+- '*me18.*'
+- 'Libraries/Boards/MAX32690/*'
+- 'Libraries/CMSIS/Device/Maxim/MAX32690/*'
+- 'Libraries/PeriphDrivers/max32690_files.mk'
+- 'Libraries/Include/MAX32690/*'
+
+'MAX78000':
+- 'Examples/MAX78000/*'
```
Comment (lorne-maxim) [inline .github/labeler.yml:148]: The MAX78000 uses some of the *me17* files.  (At least it did once upon a time.)

File: .github/workflows/PR_Title_Check.yml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+            BLE
+            Boards
+            CMSIS
+            MiscDrivers
+            PeriphDrivers
+            Third-Party
+            ignore
+            workflow
+            Other
+
```
Comment (lorne-maxim) [inline .github/workflows/PR_Title_Check.yml:60]: If these terms are going to end up in the release notes, I'd like to use "ThirdParty", "Ignore", and "Workflow" to maintain consistent capitalization and punctuation.

File: .github/workflows/PR_Title_Check.yml (hunk 1 above)
Comment (lorne-maxim) [reply .github/workflows/PR_Title_Check.yml:47]: I'm thinking we should add MAXUSB and SDHC also.  Along with BLE and Boards, these libraries are developed in house and not 3rd party.

**Review by lorne-maxim — APPROVED** [review]
I offered a few minor tweaks that you can implement if you agree, but you can also let it stand as-is.

File: .github/workflows/PR_Title_Check.yml (hunk 5 above)
Comment (lorne-maxim) [reply .github/workflows/PR_Title_Check.yml:60]: To clarify, I'm suggesting removing the '-' from "Third-Party", and capitalizing the first letter of "ignore" and "workflow".

File: .github/labeler.yml (hunk 4 above)
Comment (sihyung-maxim) [reply .github/labeler.yml:191]: You're right, so does the ME20 -> ME17 and ME16 -> ME15. I've specified the filtering based on the SVD and {part}_files.mk file.

**Pull request merged**